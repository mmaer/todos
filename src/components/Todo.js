import React from "react";
import classnames from "classnames";

import TodoInlineInput from "./TodoInlineInput";
import TodoView from "./TodoView";

const Todo = ({ editTodoTitle, toggleCompletedStatus, ...props }) => {
  return (
    <li
      className={classnames({
        completed: props.todo.completed,
        editing: props.todo.editing
      })}
    >
      <TodoView {...props} toggleCompletedStatus={toggleCompletedStatus} />
      <TodoInlineInput {...props} editTodoTitle={editTodoTitle} />
    </li>
  );
};

export default Todo;
