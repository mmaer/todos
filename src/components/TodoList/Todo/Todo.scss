.todo {
  position: relative;
  font-size: 24px;
  border-bottom: 1px solid #ededed;

  &:last-child {
    border-bottom: none;
  }
}
