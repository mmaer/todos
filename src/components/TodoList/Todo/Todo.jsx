import React from "react";

import TodoInlineInput from "./TodoEdit";
import TodoView from "./TodoView";

import "./Todo.scss";

const Todo = ({ editTodoTitle, toggleCompletedStatus, ...props }) => (
  <li className="todo">
    {props.todo.editing ? (
      <TodoInlineInput {...props} editTodoTitle={editTodoTitle} />
    ) : (
      <TodoView {...props} toggleCompletedStatus={toggleCompletedStatus} />
    )}
  </li>
);

export default Todo;
