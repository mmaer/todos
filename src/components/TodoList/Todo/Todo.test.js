import React from "react";
import { mount } from "enzyme";

import Todo from "./";

describe("Todo component", () => {
  it("Should match TodoEdit snapshot", () => {
    const wrapper = mount(<Todo editing />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Should match TodoView snapshot", () => {
    const wrapper = mount(<Todo editing={false} />);
    expect(wrapper).toMatchSnapshot();
  });
});
