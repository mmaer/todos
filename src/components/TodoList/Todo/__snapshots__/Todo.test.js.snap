// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component Should match TodoEdit snapshot 1`] = `
<Memo()
  editing={true}
>
  <li
    className="todo"
  >
    <Memo()
      editing={true}
    >
      <input
        autoFocus={true}
        className="todo-editing"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyUp={[Function]}
      />
    </Memo()>
  </li>
</Memo()>
`;

exports[`Todo component Should match TodoView snapshot 1`] = `
<Memo()
  editing={false}
>
  <li
    className="todo"
  >
    <Memo()
      editing={false}
    >
      <div
        className="todo-view"
        onDoubleClick={[Function]}
      >
        <input
          className="todo-view__toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="todo-view__label"
        />
        <button
          className="todo-view__remove"
          onClick={[Function]}
        />
      </div>
    </Memo()>
  </li>
</Memo()>
`;
